### ISSUE TITLE
#3 Model Selection, Training, and Checkpoints

### ISSUE THREAD

**Issue by SNMeans** [NeffCodes/retinopathy-risk-assessment-tool#45]
Description: Determine the model architecture, either using a pre-trained ResNet(or other CNN) model or training a new model if needed. After starting the training process, save model checkpoints to monitor process.

Tasks

- [ X] Consult with Chris/Alex to decide on using a pre-trained model or building a new CNN model.
- [ X ]  Set up and initiate the training process for the selected model.
- [ X ] Configure ```ModelCheckpoint``` to save the best-performing model based on validation loss
      -   Also ```ReduceLRonPlateau``` to adjust learning rate and ```EarlyStopping``` if no improvement after 5 steps
-  [ X ] Start model training and monitor the process for metrics like training/validation loss and accuracy.
- [ X ] Document training behavior ( plateau behavior, learning rate adjustments).

**Early Training: This is technique that stops the training process if the model stops improving on a validation metric (like accuracy or loss) after a certain number of run throughs.  This helps avoid overfitting, where the model starts to "memorize" the training data rather than learning patterns that generalize well. If a model is 'overfit', it performs much better on training data than actual testing/validation data.**

**Expected Outcome:**
The model trains successfully over 20 epochs, saving the best checkpoint and dynamically managing learning rates and early stopping.
